src/main/java/org/folio/rest/impl/EholdingsProvidersImpl.java

```diff
@@ -84,7 +89,33 @@ public void getEholdingsProviders(String q, String sort, int page, int count, Ma
 
   @Override
   public void getEholdingsProvidersByProviderId(String providerId, String include, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    asyncResultHandler.handle(Future.succeededFuture(GetEholdingsProvidersResponse.status(Response.Status.NOT_IMPLEMENTED).build()));
+    if (!headerValidator.validate(okapiHeaders, asyncResultHandler)) {
+      return;
+    }
+    CompletableFuture.completedFuture(null)
+      .thenCompose(o -> configurationService.retrieveConfiguration(new OkapiData(okapiHeaders)))
+      .thenCompose(rmapiConfiguration -> {
+        RMAPIService rmapiService = new RMAPIService(rmapiConfiguration.getCustomerId(), rmapiConfiguration.getAPIKey(),
+          rmapiConfiguration.getUrl(), vertxContext.owner());
+        return rmapiService.retrieveProvider(providerId, include);
+      })
+      .thenAccept(vendor ->
+        asyncResultHandler.handle(Future.succeededFuture(GetEholdingsProvidersByProviderIdResponse
+          .respond200WithApplicationVndApiJson(converter.convertToVendor(vendor)))))
+      .exceptionally(e -> {
+        if (e.getCause() instanceof RMAPIResourceNotFoundException) {
+          asyncResultHandler.handle(Future.succeededFuture(GetEholdingsProvidersByProviderIdResponse
+            .respond404WithApplicationVndApiJson(ErrorUtil.createError(GET_PROVIDER_NOT_FOUND_MESSAGE))));
+        } else {
+          logger.error(GET_PROVIDERS_ERROR_MESSAGE, e);
```

Comment:
I am unsure what GET_PROVIDERS_ERROR_MESSAGE is but does it have anything specific to a collection as opposed to an individual provider because then we might want to use a separate error message/tweak existing one to be generic